icepyx/core/granules.py

```diff
@@ -165,9 +180,16 @@ def get_avail(self, CMRparams, reqparams):
 
             # print(results)
 
-            if len(results["feed"]["entry"]) == 0:
-                # Out of results, so break out of loop
-                break
+            try:
+                if len(results["feed"]["entry"]) == 0:
+                    # Out of results, so break out of loop
+                    break
+            except KeyError:
+                if "errors" in results.keys():
+                    raise ValueError(
```

Comment:
Not sure if API errors fit the usual interpretation of `ValueError`. Generally I think it is good to have more specific exceptions like "CMRQueryError" or something like that so that people using the library can handle those if they want to (maybe they are doing many searches for granules and sometimes it fails and they want to continue or retry later etc...), but that could be done later.  Maybe `RuntimeError` instead, for now?

Reply:
That's a great point. I'm relatively new to raising errors, so haven't gotten beyond using built in ones yet. I've changed it to a `RuntimeError` for now, but I may see if I have a few minutes before the week is out to create a "CMRQueryError", as you suggest.

Reply:
@wallinb I added a QueryError class and tried to implement a custom error. I don't have any code to force a CMR error though - do you have an example I can use to test it?

Reply:
@JessicaS11 In general, I think sending bad input triggers an error response like this (400 status and 'errors' key in content) e.g.:
[https://cmr.earthdata.nasa.gov/search/granules.json?version=003&temporal=badinput&short_name=ATL08](https://cmr.earthdata.nasa.gov/search/granules.json?version=003&temporal=badinput&short_name=ATL08)

For writing tests, this or other error states should be mocked, e.g. can use a convenient library like [responses](https://github.com/getsentry/responses)

Now that I am looking closer at it, how about catching the error earlier with with `response.raise_for_status()`? Just adding that after the `requests.get` would result in an `HTTPError` exception thrown if there is a bad response status. If you want to do more, like raise a custom exception that prints the error message(s) CMR returns, then can do something like this:

```
try:
    response.raise_for_status()
except HTTPError as e:
    if 'errors' in response.content:  # If CMR returns a bad status with extra information, display that
        raise QueryError(response.content) from e
    else:  # If no 'errors' key, just reraise original exception
        raise
```

Reply:
I used the responses library, as you suggested, to write a test, but I have no idea if it's doing what I think it is (and it seems more like it's testing the CMR API response than it would be testing the actual lines of code in `granules.py`... or am I missing something? I've read about using mocks for testing, but have never actually written one (until now?).

The primary purpose of this PR is to address the fact that CMR would return an error but the user would never see that particular error, which was making it difficult to diagnose issues. With your sample code, I was able to implement something that seems to be doing what we want. Feel free to make any adjustment/improvements!